lib/AST/USRGeneration.cpp

```diff
@@ -135,8 +135,10 @@ bool ide::printDeclUSR(const ValueDecl *D, raw_ostream &OS) {
   } else if (isa<TypeAliasDecl>(VD) || isa<AssociatedTypeDecl>(VD)) {
     Mangler.mangleContextOf(VD);
     Mangler.mangleDeclName(VD);
-  } else {
+  } else if (!VD->isInvalid()) {
     Mangler.mangleEntity(VD, /*uncurryingLevel=*/0);
```

Comment:
Why was it failing to mangle before and why is the invalid flag the right thing to check?

If you look for usages of the invalid flag you'll see it's very unprincipled and gets set and tested in all kinds of random places. A better fix would be to either make mangleEntity more reliable or fix the code that sets up 'VD' so that it doesn't do whatever it is that made mangleEntity() crash. Otherwise, you haven't fixed the problem, you've just masked it. We might still crash later if you encounter a 'VD' that does not have isInvalid() set, but still has the same underlying structural issue.